File: src/Clients/accountant2/src/lib/models/entities/transaction.ts
import type { Syncable } from '$lib/models/syncable';
import type { Category } from '$lib/models/entities/category';

export class TransactionModel implements Syncable {
	accountName: string | undefined;
	category: Category | null = null;
	convertedAmount: number | undefined;
	synced = false;

	constructor(
		public id: number,
		public fromAccountId: number | null,
		public toAccountId: number | null,
		public categoryId: number | null,
		public amount: number,
		public fromStocks: number | null,
		public toStocks: number | null,
		public currency: string,
		public description: string | null,
		public date: string,
		public isEncrypted: boolean,
		public encryptedDescription: string | null,
		public salt: string | null,
		public nonce: string | null,
		public generated: boolean,
		public createdDate: Date,
		public modifiedDate: Date | null
	) {}

	get isTax() {
		return this.category?.isTax;
	}
}

File: src/Background/Jobs/Models/Accountant/Transaction.cs
namespace Jobs.Models.Accountant;

internal class Transaction
{
    internal int? FromAccountId { get; set; }
    internal int? ToAccountId { get; set; }
    internal int? CategoryId { get; set; }
    internal decimal Amount { get; set; }
    internal string Currency { get; set; }
    internal string Description { get; set; }
    internal DateTime Date { get; set; }
    internal bool Generated { get; set; }
    internal DateTime CreatedDate { get; set; }
    internal DateTime ModifiedDate { get; set; }
}
